sdk/program/src/lib.rs

```diff
@@ -72,3 +72,107 @@ extern crate serde_derive;
 
 #[macro_use]
 extern crate solana_frozen_abi_macro;
+
+/// Convenience macro for doing integer division where the opersation's safety
+/// can be checked at compile-time
+///
+/// Since `unchecked_div_by_const!()` is supposed to fail at compile-time, abuse
+/// doctests to cover failure modes
+/// Literal denominator div-by-zero fails
+/// ```compile_fail
+/// # use solana_program::unchecked_div_by_const;
+/// # fn main() {
+/// # let _ = unchecked_div_by_const!(10, 0);
+/// # }
+/// ```
+/// #
+/// # Const denominator div-by-zero fails
+/// ```compile_fail
+/// # use solana_program::unchecked_div_by_const;
+/// # fn main() {
+/// # const D: u64 = 0;
+/// # let _ = unchecked_div_by_const!(10, D);
+/// # }
+/// ```
+/// #
+/// # Non-const denominator fails
+/// ```compile_fail
+/// # use solana_program::unchecked_div_by_const;
+/// # fn main() {
+/// # let d = 0;
+/// # let _ = unchecked_div_by_const!(10, d);
+/// # }
+/// ```
+/// #
+/// ```compile_fail
```

Comment:
Move this case under L87? ("Literal denominator div-by-zero fails")

Reply:
These were actually "correct".  They're positioned by numerator typed (omitted from comment) then denominator type.  I did add the forgotten comments though!

Reply:
Cool, I couldn't tell if they were missing comments, or out of place. I figured moving them was slightly terser 🙂